mapper/thin_edge_json/src/json.rs

```diff
@@ -282,13 +276,109 @@ impl ThinEdgeJsonError {
 
 mod tests {
     use super::*;
+    use crate::measurement::*;
+    use assert_matches::*;
 
     fn test_timestamp() -> DateTime<FixedOffset> {
         FixedOffset::east(5 * 3600)
             .ymd(2021, 04, 08)
             .and_hms(0, 0, 0)
     }
 
+    #[test]
+    fn invalid_time_stamp() {
+        let input = r#"{
+            "time" : "2013-06-2217:03:14.000658767+02:00"
+        }"#;
+        let expected_error = r#"Invalid ISO8601 timestamp (expected YYYY-MM-DDThh:mm:ss.sss.±hh:mm): "2013-06-2217:03:14.000658767+02:00": input contains invalid characters"#;
+        let output_err = ThinEdgeJson::from_str(input, test_timestamp()).unwrap_err();
+        assert_eq!(output_err.to_string(), expected_error);
+    }
+
+    #[test]
+    fn test_parse_from_rfc3339_empty_timestamp() {
+        //no time stamp
+        assert_matches!(
+            ThinEdgeJson::parse_from_rfc3339("...."),
+            Err(ThinEdgeJsonError::InvalidTimestamp { .. })
+        );
+    }
+    #[test]
+    fn test_parse_from_rfc3339_timestamp_with_seconds() {
+        //no subseconds
+        assert_eq!(
+            ThinEdgeJson::parse_from_rfc3339("2021-04-08T00:00:00+05:00").unwrap(),
+            FixedOffset::east(5 * 3600)
+                .ymd(2021, 04, 08)
+                .and_hms(0, 0, 0)
+        );
+    }
+    #[test]
+    fn test_parse_from_rfc3339_timestamp_with_milliseconds() {
+        //with milliseconds
+        assert_eq!(
+            ThinEdgeJson::parse_from_rfc3339("2021-04-08T12:10:10.123+05:00").unwrap(),
+            FixedOffset::east(5 * 3600)
+                .ymd(2021, 04, 08)
+                .and_hms_milli(12, 10, 10, 123)
+        );
+    }
+    #[test]
+    fn test_parse_from_rfc3339_timestamp_with_nanoseconds() {
+        //with nanoseconds
+        assert_eq!(
+            ThinEdgeJson::parse_from_rfc3339("2021-04-08T12:10:10.123456789+05:00").unwrap(),
+            FixedOffset::east(5 * 3600)
+                .ymd(2021, 04, 08)
+                .and_hms_nano(12, 10, 10, 123456789)
+        );
+    }
+
+    #[test]
+    fn check_json_str_with_second_timestamp() {

```

Comment:
it'd be good to stick with a naming convention. if all your test functions start with `test_`, then better don't use `check_` as prefix for some other methods.

```rust
test_json_str_with_second_timestamp
```

is perfectly fine, except that I am missing what the test exactly is about. 

```rust
test_from_str_with_second_timestamp
```

would be better.

Reply:
I removed all the prefixes test_ or check_